src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs

```diff
@@ -0,0 +1,227 @@
+        
+        private DynamicJsonArray GetProcessThreadCollection(Process proc)
+        {
+            var dja = new DynamicJsonArray();
+            var collection = proc.Threads;
+
+            for (var idx = 0; idx < collection.Count; idx++)
+            {
+                var i = idx;
+                var djv = new DynamicJsonValue();
+
+                AddValue(djv, "Id", () => collection[i].Id);
+                AddValue(djv, "BasePriority", () => collection[i].BasePriority);
+                AddValue(djv, "CurrentPriority", () => collection[i].CurrentPriority);
+                AddValue(djv, "PriorityBoostEnabled", () => collection[i].PriorityBoostEnabled);
+                AddValue(djv, "PriorityLevel", () => collection[i].PriorityLevel);
+                AddValue(djv, "StartAddress", () => collection[i].StartAddress.ToInt64());
+                AddValue(djv, "ThreadState", () => collection[i].ThreadState);
+                AddValue(djv, "WaitReason", () => collection[i].WaitReason);
+                AddValue(djv, "PrivilegedProcessorTime", () => collection[i].PrivilegedProcessorTime);
+                AddValue(djv, "StartTime", () => collection[i].StartTime);
+                AddValue(djv, "TotalProcessorTime", () => collection[i].TotalProcessorTime);
+                AddValue(djv, "UserProcessorTime", () => collection[i].UserProcessorTime);
+                AddValue(djv, "Site", () => collection[i].Site);
+                AddValue(djv, "Container", () => collection[i].Container);
+
+                dja.Add(djv);
+            }
+            return dja;
+        }
+
+        private DynamicJsonArray GetProcessModuleCollection(Process proc)
+        {
+            var dja = new DynamicJsonArray();
+            var collection = proc.Modules;
+
+            for (var idx = 0; idx < collection.Count; idx++)
+            {
+                var i = idx;
+                var djv = new DynamicJsonValue();
+
+                AddValue(djv, "ModuleName", () => collection[i].ModuleName);
+                AddValue(djv, "FileName", () => collection[i].FileName);
+                AddValue(djv, "BaseAddress", () => collection[i].BaseAddress.ToInt64());
+                AddValue(djv, "ModuleMemorySize", () => collection[i].ModuleMemorySize);
+                AddValue(djv, "EntryPointAddress", () => collection[i].EntryPointAddress.ToInt64());
+                AddValue(djv, "FileVersionInfoFileVersion", () => collection[i].FileVersionInfo.FileVersion);
+                AddValue(djv, "Site", () => collection[i].Site);
+                AddValue(djv, "Container", () => collection[i].Container);
+
+                dja.Add(djv);
+            }
+            return dja;
+        }
+
+
+        private static void AddValue<T>(DynamicJsonValue djv, string key, Func<T> func)
+        {
+            try
+            {
+                var result = func.Invoke();
+
+                if (result == null)
+                {
+                    djv[key] = "null";
```

Comment:
Why not just set to null directly?

Reply:
https://github.com/ravendb/ravendb/pull/3763